src/Disks/S3/DiskS3.cpp

```diff
@@ -703,9 +714,11 @@ void DiskS3::removeMeta(const String & path, AwsS3KeyKeeper & keys)
             file.remove();
         }
     }
-    else
+    catch (...)
```

Comment:
This is dangerous, e.g. "memory limit exceeded".

Reply:
Maybe check for some set of error codes explicitly?

Reply:
Fixed